### PR TITLE
Feat/discussion offensive report

### PR DESCRIPTION
**Pull request by ksowndar**
Three models are created to store reports, report category, report reason
It can be controlled in admin category to view the reason of reports and can delete it
New form is added in post show page for reporting a discussion

### CODE REVIEW

File: app/admin/report_reasons.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        column :updated_at
+        actions 
+    end
+
+    form do |f|
+        f.inputs "Report Reasons" do
+          f.input :report_category,  :label => 'report_category', :as => :select, :collection => ReportCategory.all.collect {|option| [option.name, option.id]}, :selected => params[:report_category]
+          f.input :reason
+          f.input :created_at
+          f.input :updated_at
```
Comment (iMagesh) [inline app/admin/report_reasons.rb:17]: Indentations seems to be off. Can you check?

File: app/admin/reports.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+          report_category.name
+        end
+        column :reason do
+          report_reason = ReportReason.find_by(params[:report_reason_id])
+          report_reason.reason
+        end 
+        column :account
+       
+        actions 
+    end
```
Comment (iMagesh) [inline app/admin/reports.rb:18]: indentation is off here as well

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -42,9 +44,14 @@ def edit
   end
 
 
-  def draft
-    @drafts =  Post.order(created_at: :desc).page(params[:page]).per 7
-  end
+  def draft 
+    @posts=  Post.all
```
Comment (iMagesh) [inline app/controllers/posts_controller.rb:48]: Don't use the .all api, i told you before.

File: app/views/posts/show.html.erb [hunk 4]
```diff
@@ -46,7 +46,30 @@
             <%= link_to "Close Discussion", close_post_path(@post), method: :patch %>
           </div>
         <% end %>
+        <button type="button" class="join-btn" data-toggle="modal" data-target="#myModal" remote = "true">
+          Report
+       </button>
       </div>
+	</div>
```
Comment (iMagesh) [inline app/views/posts/show.html.erb:53]: check the indentation

**Review by iMagesh — CHANGES_REQUESTED** [review]
Check comments and update

File: app/controllers/posts_controller.rb (hunk 3 above)
Comment (ksowndar) [reply app/controllers/posts_controller.rb:48]: Yes, the draft methods will be corrected by @mithun30052001

File: app/admin/reports.rb (hunk 2 above)
Comment (ksowndar) [reply app/admin/reports.rb]: It is now correctly intended

File: app/views/posts/show.html.erb (hunk 4 above)
Comment (ksowndar) [reply app/views/posts/show.html.erb:53]: It is now correctly intended

**Comment by iMagesh** [comment]
@ksowndar you have to resolve conflicts

**Comment by ksowndar** [comment]
> @ksowndar you have to resolve conflicts

Ok Magesh

**Comment by ksowndar** [comment]
> @ksowndar you have to resolve conflicts

resolved @iMagesh

**Pull request merged**